### PR TITLE
Makefile targets

### PR DESCRIPTION
**Pull request by rasapala**

### CODE REVIEW

**Review by michalkulakowski — APPROVED** [review]

File: Dockerfile.redhat [hunk 1]
```diff
@@ -198,9 +198,10 @@ RUN wget -nv https://github.com/linux-test-project/lcov/releases/download/v1.16/
 COPY check_coverage.bat /ovms/
 
 ARG CHECK_COVERAGE=0
+ARG RUN_TESTS=1
 # Test Coverage
 RUN if [ "$CHECK_COVERAGE" = "1" ] ; then bazel coverage --combined_report=lcov --test_summary=detailed --test_output=streamed //src:ovms_test > ${TEST_LOG} 2>&1 || { cat ${TEST_LOG} && rm -rf ${TEST_LOG} && exit 1 ; } && genhtml --output genhtml "$(bazel info output_path)/_coverage/_coverage_report.dat" ; fi ; \
-    bazel test ${debug_bazel_flags} --jobs=$JOBS --test_summary=detailed --test_output=streamed //src:ovms_test > ${TEST_LOG} 2>&1 || (cat ${TEST_LOG} && rm -rf ${TEST_LOG} && exit 1 ; ) && tail -n 100 ${TEST_LOG} && rm -rf ${TEST_LOG}
+    if [ "$RUN_TESTS" == "1" ] ; then bazel test ${debug_bazel_flags} --jobs=$JOBS --test_summary=detailed --test_output=streamed //src:ovms_test > ${TEST_LOG} 2>&1 || (cat ${TEST_LOG} && rm -rf ${TEST_LOG} && exit 1 ; ) && tail -n 100 ${TEST_LOG} && rm -rf ${TEST_LOG} ; fi ;
```
Comment (dkalinowski) [inline Dockerfile.redhat:204]: `make docker_build RUN_TESTS=0 CHECK_COVERAGE=1` still runs unit tests. Do we want it like that? Maybe give an error when using `RUN_TESTS=0 CHECK_COVERAGE=1`? You cannot check coverage without running tests.

File: Makefile [hunk 2]
```diff
@@ -216,12 +218,16 @@ endif
 		--build-arg NVIDIA=$(NVIDIA) --build-arg ov_contrib_branch="$(OV_CONTRIB_BRANCH)" \
 		-t $(OVMS_CPP_DOCKER_IMAGE)-build:$(OVMS_CPP_IMAGE_TAG)$(IMAGE_TAG_SUFFIX) \
 		--build-arg JOBS=$(JOBS)
+
+targz_package:
```
Comment (dkalinowski) [inline Makefile:222]: `make targz_package` will fail due to missing prerequisite (ovms_build). Should we handle it somehow? Require the step? `targz_package: ovms_build`

File: Makefile [hunk 3]
```diff
@@ -216,12 +218,16 @@ endif
...
 	docker build $(NO_CACHE_OPTION) -f DockerfileMakePackage . \
 		--build-arg http_proxy=$(HTTP_PROXY) --build-arg https_proxy="$(HTTPS_PROXY)" \
-		--build-arg ov_use_binary=$(OV_USE_BINARY) --build-arg DLDT_PACKAGE_URL=$(DLDT_PACKAGE_URL) --build-arg BASE_OS=$(BASE_OS) \
+		--build-arg ov_use_binary=$(OV_USE_BINARY) --build-arg BASE_OS=$(BASE_OS) \
 		--build-arg NVIDIA=$(NVIDIA) \
 		-t $(OVMS_CPP_DOCKER_IMAGE)-pkg:$(OVMS_CPP_IMAGE_TAG) \
 		--build-arg BUILD_IMAGE=$(OVMS_CPP_DOCKER_IMAGE)-build:$(OVMS_CPP_IMAGE_TAG)$(IMAGE_TAG_SUFFIX)
+
+ovms_docker:
```
Comment (dkalinowski) [inline Makefile:230]:
```suggestion
ovms_release_image:
```

File: Makefile [hunk 4]
```diff
@@ -172,7 +173,8 @@ clang-format: venv
 	@. $(ACTIVATE); find ${STYLE_CHECK_DIRS} -regex '.*\.\(cpp\|hpp\|cc\|cxx\)' -exec clang-format-6.0 -style=file -i {} \;
 
 .PHONY: docker_build
-docker_build:
+docker_build: ovms_build targz_package ovms_docker
+ovms_build:
```
Comment (dkalinowski) [inline Makefile:177]:
```suggestion
ovms_build_image:
```

**Comment by dkalinowski** [comment]
How much time do we save when rerunning `make docker_build` ?

File: Dockerfile.redhat (hunk 1 above)
Comment (rasapala) [reply Dockerfile.redhat:204]: I think this way its is less complicated. Making additional checks and flags will not be readable. If user wants to run tests he sets the flag. If he doesn't want to run anything he sets two flags.

File: Makefile (hunk 2 above)
Comment (rasapala) [reply Makefile:222]: Ok.

File: Dockerfile.redhat (hunk 1 above)
Comment (rasapala) [reply Dockerfile.redhat:204]: Ok it works with 1 flag at the beginning so its ok.

File: Makefile [hunk 5]
```diff
@@ -172,7 +173,13 @@ clang-format: venv
 	@. $(ACTIVATE); find ${STYLE_CHECK_DIRS} -regex '.*\.\(cpp\|hpp\|cc\|cxx\)' -exec clang-format-6.0 -style=file -i {} \;
 
 .PHONY: docker_build
-docker_build:
+docker_build: ovms_builder_image targz_package ovms_release_image
+ovms_builder_image:
+ifeq ($(CHECK_COVERAGE),1)
+  ifeq ($(RUN_TESTS),0)
+	@echo "Cannot test coverage without running tetes. Use 'CHECK_COVERAGE=1 RUN_TESTS=1 make docker_build'"; exit 1 ;
```
Comment (dkalinowski) [inline Makefile:180]:
```suggestion
	@echo "Cannot test coverage without running tests. Use 'CHECK_COVERAGE=1 RUN_TESTS=1 make docker_build'"; exit 1 ;
```

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**